lib/rules/at-rule-property-required-list/index.js

```diff
@@ -30,39 +30,62 @@ const rule = (primary) => {
 			return;
 		}
 
+		/** @type {Map<string, Set<string>|undefined>} */
+		const propLists = new Map();
+
+		for (const key in primary) {
+			if (!Object.hasOwnProperty.call(primary, key)) continue;
+
+			const propList = flattenArray(primary[key]);
+
+			if (!propList) continue;
+
+			propLists.set(key, new Set(propList));
+		}
+
+		/** @type {Set<string>} */
+		const currentPropList = new Set();
+
 		root.walkAtRules((atRule) => {
 			if (!isStandardSyntaxAtRule(atRule)) {
 				return;
 			}
 
 			const { name, nodes } = atRule;
 			const atRuleName = name.toLowerCase();
-			const propList = flattenArray(primary[atRuleName]);
+			const propList = propLists.get(atRuleName);
 
 			if (!propList) {
 				return;
 			}
 
-			for (const property of propList) {
-				const propertyName = property.toLowerCase();
+			currentPropList.clear();
 
-				const hasProperty = nodes.find(
-					(node) => node.type === 'decl' && node.prop.toLowerCase() === propertyName,
-				);
+			nodes.forEach((node) => {
+				if (!node || node.type !== 'decl') return;
+
+				const propName = node.prop.toLowerCase();
+
+				if (!propList.has(propName)) return;
+
+				currentPropList.add(propName);
+			});
```

Comment:
[suggest] For consistency and readability, how about using `for...of` here like this?

```suggestion
			for (const node of nodes) {
				if (node.type !== 'decl') continue;

				const propName = node.prop.toLowerCase();

				if (!propList.has(propName)) continue;

				currentPropList.add(propName);
			}
```

Reply:
Yes, much cleaner 🙇